google-cloud-spanner/lib/google/cloud/spanner/project.rb

```diff
@@ -552,6 +552,77 @@ def client instance_id, database_id, pool: {}, labels: nil,
                      query_options: query_options
         end
 
+        ##
+        # Creates a Cloud Spanner Instance Admin client. It can be used to
+        # manage a Cloud Spanner instance.
+        #
+        # @return [Admin::Instance::V1::InstanceAdmin::Client] The newly
+        # created client.
+        #
+        # @example
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #
+        #   instance_admin_client = spanner.instance_client
+        #
+        #   project_path = \
+        #   instance_admin_client.project_path project: "my-project"
+        #   config_path = \
+        #     instance_admin_client.instance_config_path project: "my-project",
+        #   instance_config: "regional-us-central1"
+        #   job = \
+        #     instance_admin_client.create_instance parent: project_path,
+        #       instance_id: "my-instance",
+        #       instance: Google::Cloud::Spanner::Admin::Instance::V1::Instance.new({
+        #         display_name: "My Instance",
+        #         config: config_path,
+        #         node_count: 5,
+        #         labels: { "production": :env }
+        #        })
+        #   job.wait_until_done!
+        #   instance = job.results
+        #
+        def instance_client
+          Admin::Instance::V1::InstanceAdmin::Client.new do |config|
+            config.credentials = service.channel
+            config.timeout = service.timeout if service.timeout
+            config.endpoint = service.host if service.host
+            config.metadata = { "google-cloud-resource-prefix" => "projects/#{@project}" }
```

Comment:
Maybe at this point you might yield `config` to the caller of `instance_client` in case they want to do any additional configuration of the `V1::InstanceAdmin::Client` instance?